internal/store/store.go

```diff
@@ -15,10 +14,10 @@ type Repository interface {
 	AssetByID(ctx context.Context, assetID string) (*model.Asset, error)
 }
 
-func NewRepository(ctx context.Context, storeKind model.StoreKind, cfg *app.Configuration, logger *logrus.Logger) (Repository, error) {
+func NewRepository(ctx context.Context, storeKind model.StoreKind, cfg *fleetdb.Config, logger *logrus.Logger) (Repository, error) {
 	switch storeKind {
 	case model.FleetDB:
-		return fleetdb.New(ctx, cfg.FleetDBOptions, logger)
+		return fleetdb.New(ctx, cfg, logger)
 	case model.MockDB:
```

Comment:
In an age of `gomock` and `mockery` I feel like building this in is a mistake.

Reply:
Yea I think this is vestigial code. Definitely should look into removing it.

Reply:
The whole "store" idea really is odd to me. Not explained anywhere that I have found, and I generally cant find an explanation what the goal even was.

But this brings up something I have been thinking about. We duplicate these "stores" in every app. And in our controllers we have so much duplicate boilerplate. There has to be some solution to that. Crazy question I have is, why cant all of these controllers be the same app. Same configuration and setup, the only difference is what they do once you get a condition. Its more of a philosophy question i guess, but it just seems silly to me. Obviously "microservices gotta micro the services", but i feel like its too much.

Reply:
There are a bunch of different ways of doing this. Much of it comes down to "we had it working one way and decided not to change it". When I did this sort of thing at DO we had more-or-less generic workers each listening as part of a group on all the incoming message channels, and then they dispatched different message types to different go-routines.

I wouldn't want to go crazy reworking all that stuff, but a case could definitely be made for having flasher or alloy be a library that the controller app imported. I'm definitely open to improvement that you have in mind.

Reply:
Cool, but sadly its a discussion for the future, not right now. We got other things to worry about right now.